### PR TITLE
verify enrollment improvements in UI and backend

### LINKED ISSUE #5056: No Error Message when leaving Verify_Enrollment blank
### Top-level intent

Verify_Enrollment does not provide the correct response, when not entering a otp-value.

### Steps to reproduce

1. Policy->Enrollment->Verify_Enrollment: TOTP
2. Enroll TOTP Token
3. Leave OTP-Field blank and hit Verify Token

### Expected outcome

Return Message:

ERR905: The Verification of the Token has failed

### Actual outcome

Image

### Configuration

* **privacyIDEA version**: 3.12.2
* **Installation method**: apt install
* **Python version**: 3.12
* **Operating system**: Ubuntu 24
* **Webserver**: Apache2
* **Token database**: MySQL

### PR DESCRIPTION
**Pull request by nilsbehlen**
Add verify_enrollment prepolicy tests and improve token verification handling

- Added unit tests for `verify_enrollment` prepolicy, covering edge cases like missing serial, missing or incorrect verification values, and non-existent tokens.
- Refactored `verify_enrollment` to include clearer logic, early exits, and improved error handling.
- Fixed validation form in the directive to ensure user input is required and properly handled.

### CODE REVIEW

**Review by jona-samuel — APPROVED** [review]

File: tests/test_api_lib_policy.py [hunk 1]
```diff
@@ -6221,6 +6221,87 @@ def test_20_verify_enrollment(self):
...
+        with self.assertRaises(ParameterError) as cm:
+            verify_enrollment(req, None)
+
+        self.assertIn("Verification of the new token failed", str(cm.exception))
+
+        # Token should still be in verify_pending state
+        tok = get_tokens(serial=serial)[0]
+        self.assertEqual(tok.token.rollout_state, RolloutState.VERIFY_PENDING)
+        remove_token(serial)
+
```
Comment (frankmer) [inline tests/test_api_lib_policy.py:6308]: I would add test cases for scenarios with non-unique serial numbers and tokens already in the ENROLLED state to verify that the function exits early as expected.

**Review by frankmer — APPROVED** [review]
